### PR TITLE
(#234) Trial ID tests 

### LINKED ISSUE #234: Advanced Form : Trial Id section
## As a user, I want to be able to search for a trial ID

> **ESTIMATE** TBD

### Acceptance Criteria

```gherkin
Scenario: User has an option to narrow down search criteria by trial id
    Given user is navigating to "/about-cancer/treatment/clinical-trials/search/advanced"
    Then the page title is "Find NCI-Supported Clinical Trials"
    And "Trial ID" form section is displayed
    And help icon is displayed with href "/about-cancer/treatment/clinical-trials/search/help#trialid"
    And helper text "Separate multiple IDs with commas." is displayed
```
```gherkin
Scenario: User enters trial id and searches
    Given user is navigating to "/about-cancer/treatment/clinical-trials/search/advanced"
    Then the page title is "Find NCI-Supported Clinical Trials"
    And "Trial ID" form section is displayed
    When user types "NCI-2018-02825" in "TrialID" field
    When user clicks on "Find trials" button
    Then the search is executed and results page is displayed
 And trial info displays "Results 1-1  of 1 for your search "
 And the criteria table displays the following
            | Category | Selection |
  |Trial ID | NCI-2018-02825|
   And the url query has the following corresponding code
      | parameter | value          |
      | loc       | 0              |
      | rl        | 2              |
      | tid       | NCI-2018-02825 |
```

```gherkin
Scenario: User is able to search for mulriple IDs  and is able to modify search afterwards
    Given user is navigating to "/about-cancer/treatment/clinical-trials/search/advanced"
    Then the page title is "Find NCI-Supported Clinical Trials"
    And "Trial ID" form section is displayed
    When user types "NCI-2018-02825,NCI-2015-00054,NCI-2014-01507" in "TrialID" field
    When user clicks on "Find trials" button
    Then the search is executed and results page is displayed
 And trial info displays "Results 1-1  of 2 for your search "
 And the criteria table displays the following
            | Category | Selection |
  |Trial ID | NCI-2018-02825NCI-2018-02825,NCI-2015-00054,NCI-2014-01507|
    And the url query has the following corresponding code
      | parameter | value          |
      | loc       | 0              |
      | rl        | 2              |
      | tid       | NCI-2018-02825%2CNCI-2015-00054%2CNCI-2014-01507 |
   When user clicks on Modify Search Criteria button
        Then "Trial ID" field has value "NCI-2018-02825NCI-2018-02825,NCI-2015-00054,NCI-2014-01507"
        When user clears "Trial ID" input field
        And user types "NCI-2018-02825" in "Trial ID" field
        And user clicks on "Find Trials" button
        Then the search is executed and results page is displayed
 And trial info displays "Results 1-1  of 1 for your search "
 And the criteria table displays the following
            | Category | Selection |
  |Trial ID | NCI-2018-02825|
   And the url query has the following corresponding code
      | parameter | value          |
      | loc       | 0              |
      | rl        | 2              |
      | tid       | NCI-2018-02825 |
```
```gherkin
 Scenario: User has an option to go to search results url directly and modify search
        Given the user navigates to "/about-cancer/treatment/clinical-trials/search/r?loc=0&rl=2&tid=NCI-2018-02825"
        Then the page title is "Clinical Trials Search Results"
        And trial info displays "Results 1-1  of 1 for your search "
And the criteria table displays the following
            | Category | Selection |
  |Trial ID | NCI-2018-02825|
        When user clicks on Modify Search Criteria button
        Then "Trial Id" field has value "NCI-2018-02825"
```

```gherkin
 Scenario: User is searching for a trial id that does not exist will see no results found and is able to modify search
Given user is navigating to "/about-cancer/treatment/clinical-trials/search/advanced"
    Then the page title is "Find NCI-Supported Clinical Trials"
    And "Trial ID" form section is displayed
    When user types "NCI-2018-1234" in "TrialID" field
    When user clicks on "Find trials" button
    Then the search is executed and no results page is displayed
And the criteria table displays the following
            | Category | Selection |
  |Trial ID | NCI-2018-02825|
        When user clicks on Modify Search Criteria button
        Then "Trial Id" field has value "NCI-2018-1234"
  When user clears "Trial ID" input field
        And user types "NCI-2018-02825" in "Trial ID" field
        And user clicks on "Find Trials" button
        Then the search is executed and results page is displayed
 And trial info displays "Results 1-1  of 1 for your search "
 And the criteria table displays the following
            | Category | Selection |
  |Trial ID | NCI-2018-02825|
   And the url query has the following corresponding code
      | parameter | value          |
      | loc       | 0              |
      | rl        | 2              |
      | tid       | NCI-2018-02825 |
```
### Resources:
* Feature: JIRA Ticket #
* Mockups: *Here goes a URL to or the name of the mockup(s) in inVision*
* Web Page: *Here goes a URL to an existing web page*

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…

---

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- [ ] Short Spike Description - Issue #9999

### Tasks
- [ ] Short Task Description - Issue #9999
- [ ] Short Task Description - Issue #9999

### Technical Notes
- More notes, or technical nuances/direction

### PR DESCRIPTION
**Pull request by alinai**
Added feature file
Closes #234

### CODE REVIEW

File: cypress/integration/AdvancedSearchPage/TrialIdSection.feature [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	Scenario: User has an option to go to search results url directly and modify search
+		Given the user navigates to "/about-cancer/treatment/clinical-trials/search/r?loc=0&rl=2&tid=NCI-2018-02825"
+		Then the page title is "Clinical Trials Search Results"
+		And trial info displays "Results 1-1  of 1 for your search "
+		And the criteria table displays the following
+			| Category | Selection      |
+			| Trial ID | NCI-2018-02825 |
+		When user clicks on Modify Search Criteria button
+		Then "Trial Id" field has value "NCI-2018-02825"
+
```
Comment (kate-mashkina) [inline cypress/integration/AdvancedSearchPage/TrialIdSection.feature:68]: can you add this after line 67?
	```
Then "Trial ID" field has value "NCI-2018-02825"
		When user clears "TrialID" input field
		When user types "NCI-2018-02825,NCI-2015-00054,NCI-2014-01507" in "TrialID" field
		When user clicks on "Find Trials" button
		Then the search is executed and results page is displayed
		And trial info displays "Results 1-3  of 3 for your search "
		And the criteria table displays the following
			| Category | Selection                                    |
			| Trial ID | NCI-2018-02825,NCI-2015-00054,NCI-2014-01507 |
		And the url query has the following corresponding code
			| parameter | value                                        |
			| loc       | 0                                            |
			| rl        | 2                                            |
			| tid       | NCI-2018-02825,NCI-2015-00054,NCI-2014-01507 |
```

**Review by bryanpizzillo — DISMISSED** [review]
can you add a test to search for `ecog` (partial searching for one) and `ecog, swog` (two different partial searches)

File: cypress/integration/AdvancedSearchPage/TrialIdSection.feature (hunk 1 above)
Comment (alinai) [reply cypress/integration/AdvancedSearchPage/TrialIdSection.feature:81]: Done

**Comment by alinai** [comment]
> can you add a test to search for `ecog` (partial searching for one) and `ecog, swog` (two different partial searches)

Done

**Review by kate-mashkina — APPROVED** [review]

**Pull request merged**